plugins/vpp/aclplugin/data_resync_test.go

```diff
@@ -0,0 +1,489 @@
+func TestResyncErr2(t *testing.T) {
+	// Setup
+	ctx, connection, plugin := aclTestSetup(t, false)
+	defer aclTestTeardown(connection, plugin)
+
+	ctx.MockVpp.RegisterBinAPITypes(acl_api.Types)
+	ctx.MockVpp.MockReplyHandler(vppMockHandler(ctx.MockVpp, []*vppReplyMock{
+		{
+			Id:   1011,
+			Ping: true,
+			Message: &acl_api.ACLDetails{
+				ACLIndex: 0,
+				Tag:      []byte{'a', 'c', 'l', '3'},
+				Count:    1,
+				R:        []acl_api.ACLRule{{IsPermit: 1}},
+			},
+		},
+		{
+			Id:   1015,
+			Ping: true,
+			Message: &acl_api.ACLInterfaceListDetails{
+				SwIfIndex: 1,
+				Count:     2,
+				NInput:    1,
+				Acls:      []uint32{0, 2},
+			},
+		},
+		{
+			Id:   1013,
+			Ping: true,
+			Message: &acl_api.MacipACLDetails{
+				ACLIndex: 0,
+				Tag:      []byte{'a', 'c', 'l', '4'},
+				Count:    2,
+				R:        []acl_api.MacipACLRule{{IsPermit: 0}, {IsPermit: 2}},
+			},
+		},
+		{
+			Id:   1017,
+			Ping: true,
+			Message: &acl_api.MacipACLInterfaceListDetails{
+				SwIfIndex: 1,
+				Count:     1,
+				Acls:      []uint32{1},
+			},
+		},
+		{
+			Id:      1003,
+			Ping:    false,
+			Message: &acl_api.ACLDelReply{},
+		},
+		// wrong msg
+		{
+			Id:      1009,
+			Ping:    false,
+			Message: &acl_api.ACLDelReply{},
+		},
+	}))
+
+	plugin.GetL3L4AclIfIndexes().RegisterName("acl3", 1, nil)
+	plugin.GetL2AclIfIndexes().RegisterName("acl4", 1, nil)
+
+	err := plugin.Resync(acls)
+	Expect(err).To(Not(BeNil()))
+
+	// IP acl has been removed
+	Expect(len(plugin.GetL3L4AclIfIndexes().GetMapping().ListNames())).To(BeEquivalentTo(0))
+	// but MACIP acl not (wrong msg)
+	Expect(len(plugin.GetL2AclIfIndexes().GetMapping().ListNames())).To(BeEquivalentTo(1))
+
+	// old MACIP acl is still there
+	_, _, found := plugin.GetL3L4AclIfIndexes().LookupIdx("acl3")
+	Expect(found).To(BeFalse())
+	_, _, found = plugin.GetL2AclIfIndexes().LookupIdx("acl4")
+	Expect(found).To(BeTrue())
+	_, _, found = plugin.GetL3L4AclIfIndexes().LookupIdx(acls[0].AclName)
+	Expect(found).To(BeFalse())
+	_, _, found = plugin.GetL2AclIfIndexes().LookupIdx(acls[1].AclName)
+	Expect(found).To(BeFalse())
+}
+
+// Test Resync with error when configuring new ALCs
+func TestResyncErr3(t *testing.T) {
+	// Setup
+	ctx, connection, plugin := aclTestSetup(t, false)
+	defer aclTestTeardown(connection, plugin)
+
+	ctx.MockVpp.RegisterBinAPITypes(acl_api.Types)
+	ctx.MockVpp.MockReplyHandler(vppMockHandler(ctx.MockVpp, []*vppReplyMock{
+		{
+			Id:   1011,
+			Ping: true,
+			Message: &acl_api.ACLDetails{
+				ACLIndex: 0,
+				Tag:      []byte{'a', 'c', 'l', '3'},
+				Count:    1,
+				R:        []acl_api.ACLRule{{IsPermit: 1}},
+			},
+		},
+		{
+			Id:   1015,
+			Ping: true,
+			Message: &acl_api.ACLInterfaceListDetails{
+				SwIfIndex: 1,
+				Count:     2,
+				NInput:    1,
+				Acls:      []uint32{0, 2},
+			},
+		},
+		{
+			Id:   1013,
+			Ping: true,
+			Message: &acl_api.MacipACLDetails{
+				ACLIndex: 0,
+				Tag:      []byte{'a', 'c', 'l', '4'},
```

Comment:
You can convert from `string` to `[]byte` using normal brackets: `[]byte("acl4")`.